### ISSUE TITLE
Can't specify output dir

### ISSUE THREAD

**Issue by mfikes** [bhauman/figwheel-main#86]
If you use `-d` to specify the output dir as in

```
clj -Sdeps "{:deps {com.bhauman/figwheel-main {:mvn/version \"0.1.9\"}}}}"  -m figwheel.main -d out -r
```

It doesn't appear that the output directory is created and populated (with things instead being compiled in a temporary dir?)

**Comment by bhauman** [comment]
@mfikes yeah this is a bit odd.

The thing that causes this is the `should-add-temp-dir?` predicate which is a bit too simple in this case as it just checks to see if the `"target"` or `:target-dir` directory is on the classpath, if it isn't it is assumed that you aren't working on things locally and thus you don't want to output artifacts. 

A better predicate is in order I presume.

But if you do supply an output-dir with `-d` then you will also need to supply an `output-to` as well and they both will need to be in a "public" directory on the classpath otherwise the REPL isn't going to start.

If you add "target" to the classpath in a deps.edn file you will see this other behavior.

**Comment by bhauman** [comment]
```
clj -Sdeps "{:deps {com.bhauman/figwheel-main {:mvn/version \"0.1.9\"}}}  :paths [\"src\" \"target\"]}"  -m figwheel.main -d out -r
```

Should work.

**Comment by bhauman** [comment]
I could perhaps add a `--no-tmp-dir` flag?

**Comment by mfikes** [comment]
Actually my intent for specifying an output dir is that, at least for the shipping REPLS, it works around https://dev.clojure.org/jira/browse/CLJS-2917

I tried setting `-d`, saw that it didn't work, quickly checked the `figwheel.main` help output and concluded that `-d` was meant to behave like `-d` for `cljs.main`. Since it didn't work, I figured there was a simple defect and wrote this ticket.

So in terms of XY problem, I wasn't really trying to accomplish anything; just thought I had found a defect.
